commons/commons-bio/src/main/kotlin/ebi/ac/uk/io/sources/FileSourcesList.kt

```diff
@@ -23,7 +23,7 @@ import java.io.File
  *     - Open parenthesis ( ( )
  *     - Close parenthesis ( ) )
  */
-private val validPathPattern = "^(?!.*\\./)[0-9A-Za-z!-_*'(). ]+(?<!/)\$".toRegex()
+private val validPathPattern = "^(?!\\./)(?!.*/\$)(?!.*\\.\\./)[0-9A-Za-z!\\-_*'(). /]+(?!/\$)\$".toRegex()
```

Comment:
what about of:

```
private val validPathPattern = """
        ^                       # Start of the string
        (?!.*(?:\./|\.\./))     # Negative lookahead to disallow './' or '../'
        (?!.*\/$)               # Negative lookahead to disallow trailing '/'
        [0-9A-Za-z!\-_*'(). /]+ # Match one or more of the allowed characters
        $                       # End of the string
    """.trimIndent().toRegex(RegexOption.COMMENTS)
```

with some help of chat GPT

Reply:
really nice recommendation man! 👏 🤯